[WIP] Follow-ups to the CUDA framework PR review

makortel opened the pull request:
#### PR description:

This PR is to collect the follow-ups to the review of https://github.com/cms-sw/cmssw/pull/28537. Work in progress until https://github.com/cms-sw/cmssw/pull/28537 gets merged.

#### PR validation:

Code compiles, unit tests run, profiling workflow runs.

makortel commented:
@fwyzard Would it be possible to run the validation with this PR now to check the impact on performance by the last two commits? (my test on `felk40` with 8 EDM streams/threads shows little to no effect)

makortel commented:
(fixed the code format)

fwyzard commented:
:construction: Validation running at fu-c2a02-37-01:/data/user/fwyzard/patatrack/validation/run.rv8HiWsATC ...

fwyzard commented:
sure - though I suspect the throughput scan is broken, but at least I can look into what's wrong with it ...

makortel commented:
> sure - 

Thanks!

> though I suspect the throughput scan is broken, 

:(

> but at least I can look into what's wrong with it ...

:)

fwyzard commented:
Mhm, the validation failed during `git cms-merge-topic cms-patatrack:429`, more specifically during the call to `git cms-addpkg` that results in checking out 726 packages.

makortel commented:
The changes in `WaitingTaskWithArenaHolder.h` may cause the universe to be checked out (IIRC I got bitten that as well at some point). The last two commits should be independent from the others. If it helps, I can cherry-pick them to another branch/PR.

fwyzard commented:
No, there seems to be a more general issue with the validation tools and/or machines, but it's hard to pinpoint because the tests seem to work fine when run by hand.

AdrianoDee commented:
🚧 Validation running at fu-c2d41-12-01:/data/user/adiflori/patatrack-validation/run.x9gf7Uxgfw

AdrianoDee commented:
## Validation summary
Reference release [CMSSW_11_0_0_pre13](https://github.com/cms-sw/cmssw/tree/CMSSW_11_0_0_pre13) at 91be707d4e28
Development branch [cms-patatrack/CMSSW_11_0_X_Patatrack](https://github.com/cms-patatrack/cmssw/tree/CMSSW_11_0_X_Patatrack) at e41560bb9df6
Testing PRs:
  - #429 at 18f05e1ed4e5

## Validation plots
#### /RelValTTbar_13/CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/GEN-SIM-DIGI-RAW
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.5/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.5/plots_summary.html) for workflow 10824.5
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.501/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.501/plots_summary.html) for workflow 10824.501
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.502/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.502/plots_summary.html) for workflow 10824.502

#### /RelValZMM_13/CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/GEN-SIM-DIGI-RAW
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.5/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.5/plots_summary.html) for workflow 10824.5
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.501/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.501/plots_summary.html) for workflow 10824.501
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.502/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.502/plots_summary.html) for workflow 10824.502

#### /RelValTTbar_13/CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/GEN-SIM-DIGI-RAW
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/10824.5/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/10824.5/plots_summary.html) for workflow 10824.5
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/10824.501/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/10824.501/plots_summary.html) for workflow 10824.501
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/10824.502/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/10824.502/plots_summary.html) for workflow 10824.502

## Throughput plots
#### /EphemeralHLTPhysics1/Run2018D-v1/RAW run=323775 lumi=53
![scan-136.885502.png](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/EphemeralHLTPhysics1-Run2018D-v1/scan-136.885502.png)
![zoom-136.885502.png](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/EphemeralHLTPhysics1-Run2018D-v1/zoom-136.885502.png)

## logs and `nvprof`/`nvvp` profiles
#### /RelValTTbar_13/CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/GEN-SIM-DIGI-RAW
  - reference release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/reference-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/reference-10824.5-step3.log)
  - development release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.5-step3.log)
  - development release, workflow 10824.501
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.501-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.501-step3.log)
  - development release, workflow 10824.502
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-step3.log)
      - :heavy_check_mark: [profile.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-profile.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-profile.log)
      - :heavy_check_mark: `cuda-memcheck --tool initcheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-initcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-initcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool memcheck --leak-check full --report-api-errors all` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-memcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-memcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool synccheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-synccheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-synccheck.log)) did not find any errors
  - development release, workflow 136.885502
  - testing release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.5-step3.log)
  - testing release, workflow 10824.501
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.501-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.501-step3.log)
  - testing release, workflow 10824.502
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-step3.log)
      - :heavy_check_mark: [profile.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-profile.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-profile.log)
      - :heavy_check_mark: `cuda-memcheck --tool initcheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-initcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-initcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool memcheck --leak-check full --report-api-errors all` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-memcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-memcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool synccheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-synccheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-synccheck.log)) did not find any errors
  - testing release, workflow 136.885502

#### /RelValZMM_13/CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/GEN-SIM-DIGI-RAW
  - reference release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/reference-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/reference-10824.5-step3.log)
  - development release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.5-step3.log)
  - development release, workflow 10824.501
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.501-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.501-step3.log)
  - development release, workflow 10824.502
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-step3.log)
      - :heavy_check_mark: [profile.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-profile.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-profile.log)
      - :heavy_check_mark: `cuda-memcheck --tool initcheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-initcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-initcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool memcheck --leak-check full --report-api-errors all` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-memcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-memcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool synccheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-synccheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-synccheck.log)) did not find any errors
  - development release, workflow 136.885502
  - testing release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.5-step3.log)
  - testing release, workflow 10824.501
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.501-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.501-step3.log)
  - testing release, workflow 10824.502
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-step3.log)
      - :heavy_check_mark: [profile.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-profile.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-profile.log)
      - :heavy_check_mark: `cuda-memcheck --tool initcheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-initcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-initcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool memcheck --leak-check full --report-api-errors all` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-memcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-memcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool synccheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-synccheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-synccheck.log)) did not find any errors
  - testing release, workflow 136.885502

#### /RelValTTbar_13/CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/GEN-SIM-DIGI-RAW
  - reference release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/reference-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/reference-10824.5-step3.log)
  - development release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.5-step3.log)
  - development release, workflow 10824.501
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.501-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.501-step3.log)
  - development release, workflow 10824.502
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-step3.log)
      - :heavy_check_mark: [profile.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-profile.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-profile.log)
      - :heavy_check_mark: `cuda-memcheck --tool initcheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-initcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-initcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool memcheck --leak-check full --report-api-errors all` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-memcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-memcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool synccheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-synccheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-synccheck.log)) did not find any errors
  - development release, workflow 136.885502
  - testing release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.5-step3.log)
  - testing release, workflow 10824.501
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.501-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.501-step3.log)
  - testing release, workflow 10824.502
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-step3.log)
      - :heavy_check_mark: [profile.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-profile.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-profile.log)
      - :heavy_check_mark: `cuda-memcheck --tool initcheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-initcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-initcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool memcheck --leak-check full --report-api-errors all` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-memcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-memcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool synccheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-synccheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-synccheck.log)) did not find any errors
  - testing release, workflow 136.885502

## Logs
The full log is available at https://patatrack.web.cern.ch/patatrack/validation/pulls/8e22bf5430838b6e504de51b457a5665f4fd3bca/log .

makortel commented:
Thanks, so indeed no effect on the throughput.

makortel commented:
Updated to reflect the latest status of https://github.com/cms-sw/cmssw/pull/28537. In addition
* propagated the naming changes (in separate commits)
   * `{hasCUDA,requireCUDA}Devices()` -> `cms::cudatest::{test,require}Devices()`
   * `CUDAX` -> `cms::cuda::X`
   * test `CUDAX` -> `cms::cudatest::X`
   * `cudautils` -> `cms::cuda`
* removed redundant calls to `cudaDeviceCount()` (to avoid renaming them)
* rebased on top of the HEAD (corresponds merge of #430) to fix conflicts in `launch.h`

makortel commented:
Could someone relaunch the tests?

AdrianoDee commented:
🚧 Validation running at fu-c2d41-12-01:/data/user/adiflori/patatrack-validation/run.cWF6OG23Hr

AdrianoDee commented:
## Validation summary
Reference release [CMSSW_11_0_0_pre13](https://github.com/cms-sw/cmssw/tree/CMSSW_11_0_0_pre13) at 91be707d4e28
Development branch [cms-patatrack/CMSSW_11_0_X_Patatrack](https://github.com/cms-patatrack/cmssw/tree/CMSSW_11_0_X_Patatrack) at e41560bb9df6
Testing PRs:
  - #429 at 55bf9949b8c2

## Validation plots
#### /RelValTTbar_13/CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/GEN-SIM-DIGI-RAW
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.5/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.5/plots_summary.html) for workflow 10824.5
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.501/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.501/plots_summary.html) for workflow 10824.501
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.502/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.502/plots_summary.html) for workflow 10824.502

#### /RelValZMM_13/CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/GEN-SIM-DIGI-RAW
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.5/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.5/plots_summary.html) for workflow 10824.5
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.501/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.501/plots_summary.html) for workflow 10824.501
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.502/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/10824.502/plots_summary.html) for workflow 10824.502

#### /RelValTTbar_13/CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/GEN-SIM-DIGI-RAW
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/10824.5/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/10824.5/plots_summary.html) for workflow 10824.5
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/10824.501/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/10824.501/plots_summary.html) for workflow 10824.501
  - tracking validation [plots](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/10824.502/index.html) and [summary](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/10824.502/plots_summary.html) for workflow 10824.502

## Throughput plots
#### /EphemeralHLTPhysics1/Run2018D-v1/RAW run=323775 lumi=53
![scan-136.885502.png](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/EphemeralHLTPhysics1-Run2018D-v1/scan-136.885502.png)
![zoom-136.885502.png](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/EphemeralHLTPhysics1-Run2018D-v1/zoom-136.885502.png)

## logs and `nvprof`/`nvvp` profiles
#### /RelValTTbar_13/CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/GEN-SIM-DIGI-RAW
  - reference release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/reference-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/reference-10824.5-step3.log)
  - development release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.5-step3.log)
  - development release, workflow 10824.501
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.501-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.501-step3.log)
  - development release, workflow 10824.502
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-step3.log)
      - :heavy_check_mark: [profile.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-profile.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-profile.log)
      - :heavy_check_mark: `cuda-memcheck --tool initcheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-initcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-initcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool memcheck --leak-check full --report-api-errors all` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-memcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-memcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool synccheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-synccheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-synccheck.log)) did not find any errors
  - development release, workflow 136.885502
  - testing release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.5-step3.log)
  - testing release, workflow 10824.501
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.501-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.501-step3.log)
  - testing release, workflow 10824.502
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-step3.log)
      - :heavy_check_mark: [profile.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-profile.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-profile.log)
      - :heavy_check_mark: `cuda-memcheck --tool initcheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-initcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-initcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool memcheck --leak-check full --report-api-errors all` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-memcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-memcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool synccheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-synccheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-synccheck.log)) did not find any errors
  - testing release, workflow 136.885502

#### /RelValZMM_13/CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/GEN-SIM-DIGI-RAW
  - reference release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/reference-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/reference-10824.5-step3.log)
  - development release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.5-step3.log)
  - development release, workflow 10824.501
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.501-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.501-step3.log)
  - development release, workflow 10824.502
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-step3.log)
      - :heavy_check_mark: [profile.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-profile.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-profile.log)
      - :heavy_check_mark: `cuda-memcheck --tool initcheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-initcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-initcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool memcheck --leak-check full --report-api-errors all` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-memcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-memcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool synccheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-synccheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/development-10824.502-synccheck.log)) did not find any errors
  - development release, workflow 136.885502
  - testing release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.5-step3.log)
  - testing release, workflow 10824.501
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.501-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.501-step3.log)
  - testing release, workflow 10824.502
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-step3.log)
      - :heavy_check_mark: [profile.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-profile.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-profile.log)
      - :heavy_check_mark: `cuda-memcheck --tool initcheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-initcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-initcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool memcheck --leak-check full --report-api-errors all` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-memcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-memcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool synccheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-synccheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValZMM_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_realistic_v4-v1/testing-10824.502-synccheck.log)) did not find any errors
  - testing release, workflow 136.885502

#### /RelValTTbar_13/CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/GEN-SIM-DIGI-RAW
  - reference release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/reference-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/reference-10824.5-step3.log)
  - development release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.5-step3.log)
  - development release, workflow 10824.501
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.501-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.501-step3.log)
  - development release, workflow 10824.502
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-step3.log)
      - :heavy_check_mark: [profile.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-profile.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-profile.log)
      - :heavy_check_mark: `cuda-memcheck --tool initcheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-initcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-initcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool memcheck --leak-check full --report-api-errors all` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-memcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-memcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool synccheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-synccheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/development-10824.502-synccheck.log)) did not find any errors
  - development release, workflow 136.885502
  - testing release, workflow 10824.5
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.5-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.5-step3.log)
  - testing release, workflow 10824.501
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.501-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.501-step3.log)
  - testing release, workflow 10824.502
      - :heavy_check_mark: [step3.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-step3.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-step3.log)
      - :heavy_check_mark: [profile.py](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-profile.py): [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-profile.log)
      - :heavy_check_mark: `cuda-memcheck --tool initcheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-initcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-initcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool memcheck --leak-check full --report-api-errors all` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-memcheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-memcheck.log)) did not find any errors
      - :heavy_check_mark: `cuda-memcheck --tool synccheck ` ([report](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-synccheck.out), [log](https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/RelValTTbar_13-CMSSW_10_6_0-PU25ns_106X_upgrade2018_design_v3-v1/testing-10824.502-synccheck.log)) did not find any errors
  - testing release, workflow 136.885502

## Logs
The full log is available at https://patatrack.web.cern.ch/patatrack/validation/pulls/6dae33ec2217f8284c59b700b113ec6c02278297/log .

makortel commented:
Thanks. So everything still works and the performance stays the same

fwyzard commented:
there seems to be a small degradation with a larger number of streams... I will try to have a closer oook before merging.

makortel commented:
Right, I was thinking that to be noise, because the previous round of tests showed good agreement and there was no functional changes since then (only renaming stuff; ok EventCache changed to call `eventWorkHasCompleted()` instead of `cudaEventQuery()` that does add a function call, but I don't believe it could have an effect on scaling).

fwyzard commented:
OK, I rerun the scan, and it does look like noise:
![zoom](https://user-images.githubusercontent.com/4069793/72622223-166c2680-3943-11ea-9444-314702d8ec30.png)
![plot](https://user-images.githubusercontent.com/4069793/72622224-1704bd00-3943-11ea-8267-469f206f8da3.png)
